src/generic_ui/scripts/ui.ts

```diff
@@ -190,11 +191,27 @@ module UI {
      * Upon construction, the UI installs update handlers on core.
      */
     constructor(
-        public core   :uproxy_core_api.CoreApi,
+        public core   :CoreConnector,
         public browserApi :BrowserAPI) {
       // TODO: Determine the best way to describe view transitions.
       this.view = ui_constants.View.SPLASH;  // Begin at the splash intro.
 
+      core.on('connect', () => {
```

Comment:
Let's use something other than "connect" and "disconnect", as those aren't easily distinguished from login/logout or start/stop.  In this case "connect" and "disconnect" really are chrome specific and mean connect/disconnect from the app right?  In FF these just fire pretty much immediately right?